pkg/controller/storagecluster/storagecluster.go

```diff
@@ -207,6 +207,8 @@ func (c *Controller) Reconcile(ctx context.Context, request reconcile.Request) (
 
 	if err := c.validate(cluster); err != nil {
 		k8s.WarningEvent(c.recorder, cluster, util.FailedValidationReason, err.Error())
+		cluster.Status.Phase = string(corev1.ClusterOperationFailed)
+		k8s.UpdateStorageClusterStatus(c.client, cluster)
```

Comment:
return the error if the function fails.

Reply:
It is returning the original validation error. In case of update failure, let's just log it and let the original error be returned.

Reply:
@haibinxie does the change work with you?

Reply:
Yes, that sounds good.